roofit/roofitcore/inc/RooFuncWrapper.h

```diff
@@ -25,24 +26,27 @@
 #include <string>
 
 /// @brief  A wrapper class to store a C++ function of type 'double (*)(double* )'.
-/// The parameters can be accessed as x[<relative position of param in paramSet>] in the function body.
+/// The parameters can be accessed as params[<relative position of param in paramSet>] in the function body.
+/// The observables can be accessed as obs[i * num_entries + j], where i represents the observable position and j
+/// represents the data entry.
 /// @tparam Func Function pointer to the generated function.
-template <typename Func = double (*)(double *), typename Grad = void (*)(double *, double *)>
+template <typename Func = double (*)(double *, double *), typename Grad = void (*)(double *, double *, double *)>
 class RooFuncWrapper final : public RooAbsReal {
 public:
-   RooFuncWrapper(const char *name, const char *title, std::string const &funcBody, RooArgSet const &paramSet)
+   RooFuncWrapper(const char *name, const char *title, std::string const &funcBody, RooArgSet const &paramSet,
```

Comment:
Why can't the data taken by `const&` (or `const*` if you follow my other suggestion above)